File: apis_token_stateless/cadeira-livre-cliente-api/src/modulos/usuario/repository/usuarioRepository.js
import Usuario from "../model/usuario";

class UsuarioRepository {
  async findAll() {
    let usuarios = await Usuario.find();
    return usuarios;
  }

  async findById(id) {
    let usuario = await Usuario.findById(id);
    return usuario;
  }

  async findByEmail(email) {
    let usuario = await Usuario.findOne({ email });
    return usuario;
  }

  async findByCpf(cpf) {
    let usuario = await Usuario.findOne({ cpf });
    return usuario;
  }

  async save(usuario) {
    let usuarioCriado = await Usuario.create(usuario);
    return usuarioCriado;
  }
}

export default new UsuarioRepository();

File: apis_token_opaco/b2vn-auth-api/src/main/java/br/com/b2vnauthapi/b2vnauthapi/modules/usuario/repository/UsuarioRepository.java
package br.com.b2vnauthapi.b2vnauthapi.modules.usuario.repository;

import br.com.b2vnauthapi.b2vnauthapi.modules.usuario.model.Usuario;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UsuarioRepository extends JpaRepository<Usuario, Integer> {

    Page<Usuario> findById(Integer id, Pageable pageable);

    Optional<Usuario> findByEmail(String email);

    Optional<Usuario> findByCpf(String cpf);
}
